packages/api/src/clickhouse/index.ts

```diff
@@ -904,6 +912,626 @@ export const getMetricsChart = async ({
   return result;
 };
 
+export const buildMetricSeriesQuery = async ({
+  aggFn,
+  dataType,
+  endTime,
+  granularity,
+  groupBy,
+  name,
+  q,
+  startTime,
+  teamId,
+  sortOrder,
+}: {
+  aggFn: AggFn;
+  dataType: MetricsDataType;
+  endTime: number; // unix in ms,
+  granularity?: Granularity | string;
+  groupBy?: string;
+  name: string;
+  q: string;
+  startTime: number; // unix in ms
+  teamId: string;
+  sortOrder?: 'asc' | 'desc';
+}) => {
+  const tableName = `default.${TableName.Metric}`;
+  const propertyTypeMappingsModel = await buildMetricsPropertyTypeMappingsModel(
+    undefined, // default version
+    teamId,
+  );
+
+  const isRate = isRateAggFn(aggFn);
+
+  const shouldModifyStartTime = isRate;
+
+  // If it's a rate function, then we'll need to look 1 window back to calculate
+  // the initial rate value.
+  // We'll filter this extra bucket out later
+  const modifiedStartTime = shouldModifyStartTime
+    ? // If granularity is not defined (tables), we'll just look behind 5min
+      startTime - ms(granularity ?? '5 minute')
+    : startTime;
+
+  const whereClause = await buildSearchQueryWhereCondition({
+    endTime,
+    propertyTypeMappingsModel,
+    query: q,
+    startTime: modifiedStartTime,
+  });
+  const selectClause = [
+    granularity != null
+      ? SqlString.format(
+          'toUnixTimestamp(toStartOfInterval(timestamp, INTERVAL ?)) AS ts_bucket',
+          [granularity],
+        )
+      : "'0' as ts_bucket",
+    groupBy
+      ? SqlString.format(`_string_attributes[?] AS group`, [groupBy])
+      : "'' AS group",
+  ];
+
+  const hasGroupBy = groupBy != '' && groupBy != null;
+
+  if (dataType === MetricsDataType.Gauge || dataType === MetricsDataType.Sum) {
+    selectClause.push(
+      aggFn === AggFn.Count
+        ? 'COUNT(value) as data'
```

Comment:
nit: I wonder if we also want to do `toFloat64` here

Reply:
Oh good catch, we actually don't technically support the `Count` aggfn in metrics (not sure why anyone would want it either). Not sure why this is here. I'll ticket this to see if we should just clean it up later instead.